Src/VsSpecific/VsSpecific/CSharpScriptExecutor.cs

```diff
@@ -0,0 +1,172 @@
+﻿#if VS_SPECIFIC_2017 || VS_SPECIFIC_2019
+
+using System;
+using Vim.Interpreter;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+
+namespace Vim.VisualStudio.Specific
+{
+    internal sealed class CSharpScriptExecutor : ICSharpScriptExecutor
+    {
+        private const string ScriptFolder = "vsvimscripts";
+        private Dictionary<string, object> _scripts = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+
+        private MethodInfo _cSharpScriptCreate = null;
+        private dynamic _scriptOptions = null;
+        private dynamic _defaultScriptSourceResolver = null;
+        private dynamic _defaultScriptMetadataResolver = null;
+        private dynamic _defaultScriptOptions = null;
+
+        private void Execute(IVim vim, CallInfo callInfo, bool createEachTime)
+        {
+            try
+            {
+                string assemblyPath;
+                string baseDirectory = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"CommonExtensions\Microsoft\ManagedLanguages\VBCSharp\InteractiveComponents");
+
+                string assemblyName;
+                if (_cSharpScriptCreate == null)
+                {
+                    assemblyName = "Microsoft.CodeAnalysis.CSharp.Scripting.dll";
+                    assemblyPath = Path.Combine(baseDirectory, assemblyName);
+                    if (!File.Exists(assemblyPath))
+                    {
+                        vim.ActiveStatusUtil.OnError($"{assemblyName} not found.");
+                        return;
+                    }
+                    var asm = Assembly.LoadFile(assemblyPath);
+                    var t = asm.GetType("Microsoft.CodeAnalysis.CSharp.Scripting.CSharpScript");
+                    foreach (MethodInfo mi in t.GetMethods())
+                    {
+
+                        if (!mi.IsGenericMethod && mi.Name == "Create")
+                        {
+                            if (mi.GetParameters()[0].ParameterType == typeof(string))
+                            {
+                                _cSharpScriptCreate = mi;
+                                break;
+                            }
+                        }
+                    }
+                    if(_cSharpScriptCreate == null)
+                    {
+                        vim.ActiveStatusUtil.OnError($"method 'CSharpScript.Create' not found.");
+                        return;
+                    }

```

Comment:
Rather than having all this reflection why not add references to the C# scripting API directly. If we make sure to reference the 1.0 version, which is what you're logically doing here, those will load fine in later versions of Visual Studio. Because it's a supported API set the have the correct binding redirects in the box.

Reply:
As you advised, I did not need to do something like this.
After deciding what to do with ProcessFromScript, I would like to fix it.